src/Razor/src/Microsoft.AspNetCore.Razor.Language/Syntax/StackGuard.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.Razor.Language.Syntax
+{
+    internal static class StackGuard
```

Comment:
Unrelated to this PR. Added this from Roslyn. This gives a meaningful exception with stacktrace `InsufficientExecutionStackException` rather than abruptly exiting with a Stackoverflow in extremely nested cases.